src/Exception/ConnectException.php

```diff
@@ -10,22 +12,32 @@
  *
  * Note that no response is present for a ConnectException
  */
-class ConnectException extends RequestException implements NetworkExceptionInterface
+class ConnectException extends TransferException implements NetworkExceptionInterface
 {
+    use RequestAwareTrait, HandlerContextAwareTrait;
+
     public function __construct(
         string $message,
         RequestInterface $request,
         \Throwable $previous = null,
         array $handlerContext = []
     ) {
-        parent::__construct($message, $request, null, $previous, $handlerContext);
+        parent::__construct($message, 0, $previous);
+        $this->setRequest($request);
+        $this->setHandlerContext($handlerContext);
     }
 
+    /**
+     * @deprecated 8.0.0
```

Comment:
We can't mark this as deprecated until the 7.1.0 release, so these annotations will need to come in a followup PR after this one, if this is merged into what will be 7.0.0.